### DISCUSSION TITLE
Roadmap for 2022

### DISCUSSION THREAD

**Discussion by olimorris** [olimorris/onedarkpro.nvim#82] [Ideas]
There have been lot's of great ideas recently which are finding their way over into the [issues](https://github.com/olimorris/onedarkpro.nvim/issues) tab, so I wanted to publicise what my plans are for the remainder of 2022 and open them up for discussion.

## Planned Roadmap

1. Compilation - As pioneered by [Nightfox](https://github.com/EdenEast/nightfox.nvim), the ability to compile your onedarkpro setup into a file for faster boot times
2. Filetype Highlights - [Changes](https://github.com/neovim/neovim/issues/14090#issuecomment-1228444035) for defining Treesitter highlight groups in Nightly (Neovim 0.8) make this feature more exciting. I plan on adding a layer of abstraction to the config to make it easier for users to set TS highlights, segmented by filetype
3. Opinionated Filetype Highlights - Opinionated highlights baked into the color scheme (with the ability to toggle on/off) to get closer to the original One Dark Pro theme when loading specific filetypes#81 

## Principles

[To be updated]

**Comment by mmirus** [comment]
Sounds awesome! If the upstream work on semantic tokens drops at some point, my impression is that could also enable some extra polish for specific file types. Not sure if/when that might happen though. 

Looking forward to following along with this!

**Reply by olimorris** [reply]
Could you share a link to that? Sounds very interesting

**Reply by mmirus** [reply]
This is probably a good jumping off point: https://github.com/neovim/neovim/pull/15723.

And https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens

**Comment by mrjones2014** [comment]
Some ideas I had for your 3rd point, specifically around opinionated Markdown highlighting:

- `# Headings` get bolded and a unique text color (possibly separate colors for different nesting levels?)
- `*inline bold*` gets bolded
- `**inline italics**`/`_inline italics_` get italicized
- `inline code` gets a background color and a brighter/non-`Normal` foreground color

**Reply by mrjones2014** [reply]
Also maybe for `[links](https://example.com)` and `![image alt text](https://example.com/image.png)`, the whole thing gets italicized and the URL gets underlined.

**Comment by olimorris** [comment]
Happy to report that I've added **caching** support to the colorscheme. I'd previously called this compilation but caching grew on me over time. Please let me know what you think!

Filetype highlights will be next to tackle and I'd appreciate your input.
